jupyter-spark/spark.py

```diff
@@ -18,17 +19,45 @@ def raise_error(msg):
 
 class SparkHandler(IPythonHandler):
     spark_host = None
+    web_app = None
 
     def get(self):
         #tornado_logger.info("Received URI from client: " + self.request.uri)
         if not self.request.uri.startswith(EXTENSION_URL):
             raise_error("URI did not start with " + EXTENSION_URL)
-        spark_request = spark_host + self.request.uri[len(EXTENSION_URL):]
+        spark_request = self.spark_host + self.request.uri[len(EXTENSION_URL):]
         #tornado_logger.info("Sending request to Spark UI: " + spark_request)
 
         try:
             spark_response = requests.get(spark_request)
-            client_response = spark_response.text
+
+            content_type = spark_response.headers['content-type']
+            self.set_header("Content-Type", content_type)
+
+            if "text" in content_type:
+                # Replace all the relative links with our proxy links
+                soup = BeautifulSoup(spark_response.text)
+
+                for has_href in ['a', 'link']:
+                    for a in soup.findAll(has_href):
+                        try:
+                            a['href'] = url_path_join(self.web_app, a['href'])
+                        except KeyError:
+                            print "boo", a
```

Comment:
These print statements are python2-style. We should change them to `tornado_logger.info()` calls if they're useful user-facing error logging, and remove them otherwise.